docs/book/driver-deployment/installation.md

```diff
@@ -0,0 +1,226 @@
+user = "<username>"
+password = "<password>"
+port = "<port>"
+ca-file = <ca file path> # optional, use with insecure-flag set to false
+datacenters = "<datacenter-name1>, <datacenter-name2>, ..."
+```
+
+Where the entries have the following meaning:
+
+- `cluster-id` - represents the unique cluster identifier. Each kubernetes cluster should have it's own unique cluster-id set in the configuration file.
+
+- `VirtualCenter` - section defines vCenter IP address / FQDN.
+
+- `insecure-flag` - should be set to true to use self-signed certificate for login.
+
+- `user` - vCenter username.
+
+- `password` - password for vCenter user specified with user.
+
+- `port` - vCenter Server Port. The default is 443 if not specified.
+
+- `ca-file` - path to a CA certificate in PEM format. It is an optional parameter.
+
+- `datacenters` - list of all comma separated datacenters where kubernetes node VMs are present.
+
+### vSphere configuration file for file volumes
+
+For file volumes, there are some extra parameters added to the config to help with the network permissions and placement of volumes. A sample config file for file volumes is shown below.
+
+```cgo
+$ cat /etc/kubernetes/csi-vsphere.conf
+[Global]
+cluster-id = "<cluster-id>"
+
+[NetPermissions "A"]
+ips = "*"
+permissions = "READ_WRITE"
+rootsquash = false
+
+[NetPermissions "B"]
+ips = "10.20.20.0/24"
+permissions = "READ_ONLY"
+rootsquash = true
+
+[NetPermissions "C"]
+ips = "10.30.30.0/24"
+permissions = "NO_ACCESS"
+
+[NetPermissions "D"]
+ips = "10.30.10.0/24"
+rootsquash = true
+
+[NetPermissions "E"]
+ips = "10.30.1.0/24"
+
+[VirtualCenter "<IP or FQDN>"]
+insecure-flag = "<true or false>"
+user = "<username>"
+password = "<password>"
+port = "<port>"
+ca-file = <ca file path> # optional, use with insecure-flag set to false
+datacenters = "<datacenter-name1>, <datacenter-name2>, ..."
+targetvSANFileShareDatastoreURLs = "ds:///vmfs/volumes/vsan:52635b9067079319-95a7473222c4c9cd/" # Optional
+```
+
+Some of the parameters have been explained in the previous section for block volumes.  
+
+`targetvSANFileShareDatastoreURLs` and `NetPermissions` section are exclusive to file volumes and are optional.
+
+- `targetvSANFileShareDatastoreURLs` - optional parameter. It comes in handy when you have an environment with file service enabled vSAN cluster(s) and you intend to limit the creation of file share volumes to only a select few vSAN datastores. This field contains a comma separated list of datastore URLs where you want to deploy the file share volumes.
+
+- `NetPermissions` - optional parameter. Set of parameters used to restrict the network capabilities of all the file share volumes created under this vSphere configuration. If the complete set of `NetPermissions` are not mentioned for a given IP range, defaults are assumed for the missing parameters. You can define as many `NetPermissions` sections as you want and each of these sections is uniquely identified by the string which follows.
```

Comment:
Can we specify what is the default if `NetPermissions` is not specified.

Reply:
@RaunakShah Could you add the line below to the end of this paragraph.
"If the `NetPermissions` section is completely omitted, the defaults for each of the parameters below are assumed."